internal/storage/auth/cache/cache.go

```diff
@@ -0,0 +1,114 @@
+package cache
+
+import (
+	"context"
+	"fmt"
+
+	"go.flipt.io/flipt/internal/cache"
+	"go.flipt.io/flipt/internal/storage/auth"
+	authrpc "go.flipt.io/flipt/rpc/flipt/auth"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type Store struct {
+	auth.Store
+	cacher cache.Cacher
+	logger *zap.Logger
+}
+
+const (
+	// storage:auth:token:<token>
+	// nolint:gosec
+	authTokenCacheKeyFmt = "s:a:t:%s"
+	// storage:auth:id:<id>
+	authIDCacheKeyFmt = "s:a:i:%s"
+)
```

Comment:
the idea is we keep an index in the cache:

auth id -> token

since we want to be able to delete the token from the cache in case of an expiration call, however the only expiration call that we allow is `ExpireAuthByID`

so we need to have a way to associate a token with its ID